index.bs

```diff
@@ -1434,7 +1285,7 @@ interface MLGraphBuilder {
 </script>
 
 <div class="note">
-Both {{MLGraphBuilder}}.{{MLGraphBuilder/build()}} and {{MLGraphBuilder}}.{{MLGraphBuilder/buildSync()}} methods compile the graph builder state up to the specified output operands into a compiled graph according to the type of {{MLContext}} that creates it. Since this operation can be costly in some machine configurations, the calling thread of the {{MLGraphBuilder}}.{{MLGraphBuilder/buildSync()}} method must only be a worker thread to avoid potential disruption of the user experience. When the {{[[contextType]]}} of the {{MLContext}} is set to "[=context type/default=]", the compiled graph is initialized right before the {{MLGraph}} is returned. This graph initialization stage is important for optimal performance of the subsequent graph executions. See [[#api-mlcommandencoder-graph-initialization]] for more detail.
+Both {{MLGraphBuilder}}.{{MLGraphBuilder/build()}} and {{MLGraphBuilder}}.{{MLGraphBuilder/buildSync()}} methods compile the graph builder state up to the specified output operands into a compiled graph according to the type of {{MLContext}} that creates it. Since this operation can be costly in some machine configurations, the calling thread of the {{MLGraphBuilder}}.{{MLGraphBuilder/buildSync()}} method must only be a worker thread to avoid potential disruption of the user experience. When the {{[[contextType]]}} of the {{MLContext}} is set to "[=context type/default=]", the compiled graph is initialized right before the {{MLGraph}} is returned. This graph initialization stage is important for optimal performance of the subsequent graph executions.
```

Comment:
Graph initialization was specified along with the `MLCommandEncoder`, and I've proposed removing it in this PR. Is this something we need to concretely specify? Or should the concept of graph initialization be a non-normative hint?

> When the {{[[contextType]]}} of the {{MLContext}} is set to "[=context type/default=]", the compiled graph is initialized right before the {{MLGraph}} is returned

This is normative text, so if this is true, graph initialization should be specified and used in the algorithm for `buildSync()`. It looks like `buildSync()` includes steps to "initialize the graph", but it does so without regard for the "content type". Should the algorithm be updated? And/or this text?

Reply:
I'd move the graph init as a separate algorithm, used/referred by the buildSync() steps.

Reply:
Given that graph init is not specified, I say we file an issue to specify graph initialization and link to that issue from this text here. WDYT?

Reply:
> Opening an issue for graph initialization sounds good.

Filed #552!

> For this PR, I recommend moves the graph initialization into a non-normative note merged with the note of `MLCommandEncoder::initializeGraph`.

SGTM. Added to the existing "note" for `MLGraphBuilder`, with a link to #552. I expect we may want to move this text around when we address #552 

> > Is the "context type" relevant to that algorithm, as the text here suggests?
> 
> Should we also remove [`createContext(gpuDevice)`](https://www.w3.org/TR/webnn/#dom-ml-createcontext-gpudevice) and [`MLBufferResourceView`](https://www.w3.org/TR/webnn/#ref-for-dictdef-mlbufferresourceview)? I think they are specified for previous WebGPU interop.

Hmm good question... I say let's punt on these for the sake of keeping PRs small and relatively self-contained. I also think there's still a change we want the ability to create an MLContext associated with a GPUDevice - say, to know where to allocate an `MLBuffer` - so I don't want to preemptively remove text we might still use. That being said, `MLBufferResourceView` seems safe to go. I can put up a follow-up once this PR merges

Reply:
> That being said, `MLBufferResourceView` seems safe to go. I can put up a follow-up once this PR merges

There appear to be no merge conflicts, so I went ahead and put up #553

Reply:
> I also think there's still a change we want the ability to create an MLContext associated with a GPUDevice - say, to know where to allocate an MLBuffer - so I don't want to preemptively remove text we might still use.

It makes sense.

Reply:
Resolving this thread since issues have been filed for everything discussed here